src/nsls2api/services/proposal_service.py

```diff
@@ -581,6 +581,30 @@ async def synchronize_proposal_from_pass(proposal_id: int) -> None:
     logger.debug(f"Response: {response}")
 
 
+async def update_proposals_with_cycle(cycle_name: str) -> None:
+    """
+    Update the cycle <-> proposals mapping for the given cycle.
+
+    :param cycle_name: The name of the cycle to process proposals for.
+    :type cycle_name: str
+    """
+
+    proposal_list = await fetch_proposals_for_cycle(cycle_name)
+
+    logger.info(f"Found {len(proposal_list)} proposals for cycle {cycle_name}.")
+
+    for proposal_id in proposal_list:
+        # Add the cycle to the Proposal object
+
+        try:
+            proposal = await proposal_by_id(int(proposal_id))
+            await proposal.update(AddToSet({Proposal.cycles: cycle_name}))
```

Comment:
Does this need to be an array -- to make it append to a list of cycles?

```suggestion
            await proposal.update(AddToSet({Proposal.cycles: [cycle_name]}))
```

I don't know beanie, thought it seemed to reference the mongodb operator of the same name...
https://www.mongodb.com/docs/manual/reference/operator/update/addToSet/#value-to-add-is-an-array

Reply:
No it doesn't need to be an array in python